### DISCUSSION TITLE
Why litre dimensionality is length and not volume?

### DISCUSSION THREAD

**Discussion by felipythondev** [hgrecco/pint#1975] [Q&A] [answered]
I'm working on setting a feature to check the dimensionality of some units of measure, but I get length as the dimensionality for litre, ml as well. Why is that? Should it be volume? How could I get volume as unit of measure for it?

**Comment by dopplershift** [comment, marked as answer]
```python
import pint

ureg = pint.UnitRegistry()
x = 3 * ureg('mL')
x.dimensionality
```
gives `<UnitsContainer({'[length]': 3})>`, which says the dimensionality is length**3, which implies volume?

**Reply by felipythondev** [reply]
Hmm, got you I was expecting it to be something like `<UnitsContainer({'[volume]': 1})>`

**Comment by andrewgsavage** [comment]
Have a play with https://github.com/hgrecco/pint/pull/1967 It needs reviewing and feedback.
With that PR,
```python3
import pint

ureg = pint.UnitRegistry()
x = 3 * ureg('mL')
x.compatible_kinds()
{'[volume]'}
```

> I'm working on setting a feature  

Could you expand on this? Understanding how you intend to use this could help make test cases.
